src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import './TodoList.css';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { LoadingType } from '../../types';
+
+interface Props {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  loadingIds: LoadingType;
+  onEdit: (
+    todo: Todo,
+    key: keyof Todo,
+    value: boolean | string,
+  ) => Promise<boolean>;
+  onDelete: (todoID: number) => Promise<void>;
+}
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  tempTodo,
+  loadingIds,
+  onEdit,
+  onDelete,
+}) => (
+  <section className="todoapp__main" data-cy="TodoList">
+    <TransitionGroup>
+      {todos.map(todo => (
+        <CSSTransition key={todo.id} timeout={300} classNames="item">
+          <TodoItem
+            key={todo.id}
```

Comment:
better use uuid library

Reply:
I tried to make the key with uuid, but the test started failing. So I deleted CSSTransition, but it didn`t help, so I just left this part without changes, but I will use it in future projects )

Reply:
okay no problem